modules/core/src/test/java/org/apache/ignite/cdc/WalForCdcTest.java

```diff
@@ -0,0 +1,250 @@
+    public void testOnlyDataRecordWritten() throws Exception {
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        AtomicInteger cntr = new AtomicInteger();
+
+        // Check only `DataRecords` written in WAL for in-memory cache with CDC enabled.
+        doTestWal(ignite, cache -> {
+            for (int i = 0; i < RECORD_COUNT; i++)
+                cache.put(keyForNode(ignite.affinity(DEFAULT_CACHE_NAME), cntr, ignite.localNode()), i);
+        });
+
+        // Check no WAL written during rebalance.
+        IgniteEx ignite1 = startGrid(1);
+
+        awaitPartitionMapExchange(false, true, null);
+
+        // Can't use `waitForCondition` because if test passed
+        // then no `DataRecords` loged therefore no segment archivation.
+        Thread.sleep(3 * WAL_ARCHIVE_TIMEOUT);
+
+        int walRecCnt = checkDataRecords(ignite1);
+
+        assertEquals(0, walRecCnt);
+
+        // Check `DataRecords` written on second node after rebalance.
+        doTestWal(ignite1, cache -> {
+            for (int i = 0; i < RECORD_COUNT; i++)
+                cache.put(keyForNode(ignite1.affinity(DEFAULT_CACHE_NAME), cntr, ignite1.localNode()), i);
+        });
+    }
+
+    /** */
+    @Test
+    public void testWalDisable() throws Exception {
+        persistenceEnabled = true;
+
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        doTestWal(ignite, cache -> {
+            for (int i = 0; i < RECORD_COUNT / 2; i++)
+                cache.put(i, i);
+
+            ignite.cluster().disableWal(DEFAULT_CACHE_NAME);
+
+            for (int i = 0; i < RECORD_COUNT; i++)
+                cache.put(i, i);
+
+            ignite.cluster().enableWal(DEFAULT_CACHE_NAME);
+
+            for (int i = RECORD_COUNT / 2; i < RECORD_COUNT; i++)
+                cache.put(i, i);
+        });
+    }
+
+    /** */
+    @Test
+    public void testWalDisabledIfPersistenceAndCdcDisabled() throws Exception {
+        persistenceEnabled = false;
+        cdcEnabled = false;
+
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        ignite.getOrCreateCache(new CacheConfiguration<Integer, Integer>(DEFAULT_CACHE_NAME)
+                .setCacheMode(mode)
+                .setAtomicityMode(atomicityMode));
+
+        assertNull(ignite.context().cache().context().wal());
+        assertNull(getFieldValue(ignite.context().cache().context(), "cdcWalMgr"));
+    }
+
+    /** */
+    private void doTestWal(IgniteEx ignite, Consumer<IgniteCache<Integer, Integer>> putData) throws Exception {
+        IgniteCache<Integer, Integer> cache = ignite.getOrCreateCache(
+            new CacheConfiguration<Integer, Integer>(DEFAULT_CACHE_NAME)
+                .setCacheMode(mode)
+                .setAtomicityMode(atomicityMode));
+
+        long archiveIdx = ignite.context().cache().context().wal(true).lastArchivedSegment();
```

Comment:
WDYT, if we add default method `wal()` --> `wal(false)`?

Reply:
Method added.